client/my-sites/comments/comment-list/index.jsx

```diff
@@ -90,16 +94,29 @@ export class CommentList extends Component {
 		this.props.removeNotice( `comment-notice-${ commentId }` );
 
 		this.props.deleteComment( commentId, postId );
-	}
+	};
 
 	editComment = ( commentId, postId, commentData, undoCommentData, showNotice = true ) => {
 		this.props.editComment( commentId, postId, commentData );
 		if ( showNotice ) {
 			this.showEditNotice( commentId, postId, undoCommentData );
 		}
-	}
+	};
 
-	getComments = () => uniq( [ ...this.state.persistedComments, ...this.props.comments ] ).sort( ( a, b ) => b - a );
+	getComments = () => {
+		const comments = uniq( [ ...this.state.persistedComments, ...this.props.comments ] );
+
+		// While sorting we are relying on the fact that newer comments will have larger ID values.
+		switch ( this.state.sortOrder ) {
+			case NEWEST_FIRST:
+				return comments.sort( ( a, b ) => b - a );
+			case OLDEST_FIRST:
+				return comments.sort( ( a, b ) => a - b );
+			default:
+				// Default to showing newest comments first.
+				return comments.sort( ( a, b ) => b - a );
+		}
+	};
```

Comment:
We could simplify this with `_.orderBy`:

```js
const { sortOrder } = this.state;
return orderBy( comments, null, OLDEST_FIRST === sortOrder ? 'asc' : 'desc' );
```

Reply:
I was tinkering with that approach too, but left it like this in case we wanted to add other sorting methods in the future. On the other hand, since we can only compare ids here, additional sorting methods will probably require a more involved refactor, and at that point we'll probably have to rewrite most of it anyway - so I'm going to simplify it as you've suggested. :)

Actually, I was already prepared for this and _wisely_ used `asc` and `desc` for corresponding constant names, so we can simplify it even further to:
```javascript
return orderBy( comments, null, this.state.sortOrder );
```
:smile: 

![choosewisely](https://user-images.githubusercontent.com/1182160/30753202-e584ca6a-9fbe-11e7-8063-a0b445c2881a.jpg)

Reply:
![choosewisely](https://media.giphy.com/media/ZgYBhq1x7L1bW/200.gif)